matrix/Matrix.hpp

```diff
@@ -288,6 +288,29 @@ class Matrix
         *this = (*this) - scalar;
     }
 
+    bool operator==(const Matrix<Type, M, N> &other) const
+    {
+        const Matrix<Type, M, N> &self = *this;
+
+        // TODO: set this based on Type
+        static constexpr float eps = 1e-4f;
```

Comment:
Why not make `eps` an optional parameter with default? So you can customize the expected accuracy as client?
That's already done in several places:
- https://github.com/PX4/Matrix/blob/f835d390171ae61de977d00cdaf5ed16e5361e4e/matrix/Vector.hpp#L91
- https://github.com/PX4/Matrix/blob/41ad2bdea52771a8f486a0be6098e6e45786bdcd/matrix/Matrix.hpp#L505
- https://github.com/PX4/Matrix/blob/f835d390171ae61de977d00cdaf5ed16e5361e4e/matrix/Quaternion.hpp#L197

Reply:
I wasn't sure of a good way to do that via operator overloading. You can't pass additional arguments. With templates you can use enable_if to have different things for different types, but it gets ugly quickly.

Reply:
Then lets add that option to isEqual via argument when we merge them or find an even better solution based on types.